indexing-service/src/main/java/org/apache/druid/indexing/common/tasklogs/ConsoleLoggingEnforcementConfigurationFactory.java

```diff
@@ -117,14 +117,15 @@ private Appender findConsoleAppender()
     }
 
     /**
-     * remove all appenders from a logger and append a console appender to it
+     * Ensure there is a console logger defined. Without a console logger peon logs wont be able to be stored in deep storage
      */
     private void applyConsoleAppender(LoggerConfig logger, Appender consoleAppender)
     {
-      if (logger.getAppenderRefs().size() == 1
-          && logger.getAppenderRefs().get(0).getRef().equals(consoleAppender.getName())) {
-        // this logger has only one appender and its the console appender
-        return;
+      for (AppenderRef appenderRef : logger.getAppenderRefs()) {
+        if (consoleAppender.getName().equals(appenderRef.getRef())) {
+          // we need a console logger no matter what, but we want to be able to define a different appender if necessary
+          return;
+        }
       }
 
```

Comment:
Can we add some way of notifying the Druid operator that their log4j appenderator config is being overwritten because they are missing a console appender. The notice should tell the operator that if they want their config to stick, they must also add the console appender to the logger